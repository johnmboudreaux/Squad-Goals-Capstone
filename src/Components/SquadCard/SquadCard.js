import React from 'react';
import PropTypes from 'prop-types';

const SquadCard = ({ squad }) => {
  return (
    <tr className="squad-row">
      <td>{squad.squad_name}</td>
      <td>{squad.id}</td>
      <td>{squad.id}</td>
      <td><input type="button" value="Leave" /></td>
    </tr>
  );
};

SquadCard.propTypes = {
  squad: PropTypes.object,
};

export default SquadCard;
