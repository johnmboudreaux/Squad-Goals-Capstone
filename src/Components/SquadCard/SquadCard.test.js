/* eslint-disable */
import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import SquadCard from './SquadCard';
import squad from '../../../data/test/squad_data_test';

describe('SquadCard snapshot', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<SquadCard
      squad={squad}
    />);

    expect(wrapper).toMatchSnapshot();
  });


  // it('renders without crashing', () => {
  //   const div = document.createElement('div');
  //   ReactDOM.render(<SquadCard
  //     squad={squad}
  //   />, div);
  // });
});
