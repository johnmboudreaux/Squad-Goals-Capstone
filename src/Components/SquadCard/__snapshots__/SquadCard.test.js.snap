// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquadCard snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SquadCard
    squad={
      Array [
        Object {
          "conversation_id": 5,
          "id": 1,
          "squad_name": "Gizmo",
        },
        Object {
          "conversation_id": 6,
          "id": 2,
          "squad_name": "Mickey",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          <Link
            replace={false}
            to={
              Object {
                "pathname": "/Squad/undefined",
                "state": Object {
                  "userId": undefined,
                },
              }
            }
          />
        </td>,
        <td />,
        <td />,
      ],
      "className": "squad-row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to={
              Object {
                "pathname": "/Squad/undefined",
                "state": Object {
                  "userId": undefined,
                },
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "replace": false,
            "to": Object {
              "pathname": "/Squad/undefined",
              "state": Object {
                "userId": undefined,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/Squad/undefined",
                  "state": Object {
                    "userId": undefined,
                  },
                }
              }
            />
          </td>,
          <td />,
          <td />,
        ],
        "className": "squad-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to={
                Object {
                  "pathname": "/Squad/undefined",
                  "state": Object {
                    "userId": undefined,
                  },
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "replace": false,
              "to": Object {
                "pathname": "/Squad/undefined",
                "state": Object {
                  "userId": undefined,
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
